public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -105,6 +105,15 @@ export class ElasticDatasource extends DataSourceApi<ElasticsearchQuery, Elastic
     }
   }
 
+  private requestAllIndices(indexList: string[], index: number, url: string): Promise<any> {
+    return this.request('GET', indexList[index] + url).catch(err => {
```

Comment:
Lets change this to a for cycle ie something like:
```
for (const index of indexList) {
  try {
    return await request(index)
  } catch (error) {
     // continue
  }
}
```
Usually I prefer using for/while cycles instead of recursion (though there are some algorithms which are more readable in recursion form which would made the exceptions), also tail call optimisation isn't common in js engines (even though here probably not a problem).

Reply:
Thanks, that's exactly what I'm looking for!